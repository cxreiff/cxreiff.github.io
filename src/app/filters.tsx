import { FC } from 'react'

export const Filters: FC = () => {
    return (
        <>
            <Turbulence />
        </>
    )
}

const Turbulence = () => (
    <svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="0" height="0">
        <defs>
            <filter id="turbulence_0">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="0" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_1">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="1" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_2">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="2" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_3">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="3" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_4">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="4" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_5">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="5" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_6">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="6" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_7">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="7" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
        </defs> 
    </svg>
)
