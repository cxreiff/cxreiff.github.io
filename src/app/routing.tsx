import { FC, ReactNode, Suspense, useEffect, lazy } from "react";
import { Routes, Route, useLocation } from "react-router-dom";

import { Loader } from "src/common/loader/loader.component";
import { IFrame } from "src/common/iframe/iframe.component";
import {
  DOCUMENT_TITLE_ROOT,
  BREAKOUT_URL,
  PLATFORMER_URL,
  HEX_TILES_URL,
  PIXIJS_URL,
  BABYLONJS_URL,
  PLAIN_WEBGL_URL,
  BIGFOOT_MAP_URL,
} from "src/utilities/constants";

import Homepage from "src/features/homepage/homepage.component";
import Projects from "src/features/projects/projects.component";
import Photos from "src/features/photos/photos.component";
import Sketches from "src/features/sketches/sketches.component";
import ProjectsFigrid from "src/features/projects/projectsFigrid.component";

// const Posts = lazy(() => import("src/features/posts/posts.component"));
// const PostsContent = lazy(
//   () => import("src/features/posts/postsContent.component")
// );

const Shaders = lazy(() => import("src/features/shaders/shaders.component"));

const Pixijs = () => (
  <IFrame aspect={15.5 / 9} url={PIXIJS_URL} title={"pixijs"} canFullscreen />
);
const Babylonjs = () => (
  <IFrame
    aspect={3 / 2}
    url={BABYLONJS_URL}
    title={"babylonjs"}
    canFullscreen
  />
);
const Breakout = () => (
  <IFrame
    aspect={3 / 2}
    mobileAspect={2 / 3}
    url={BREAKOUT_URL}
    title={"breakout"}
    canFullscreen
  />
);
const Platformer = () => (
  <IFrame
    aspect={16 / 10}
    url={PLATFORMER_URL}
    title={"platformer"}
    canFullscreen
  />
);
const HexTiles = () => (
  <IFrame
    aspect={16 / 10}
    url={HEX_TILES_URL}
    title={"hex tiles"}
    canFullscreen
  />
);
const PlainWebGL = () => (
  <IFrame
    aspect={16 / 10}
    url={PLAIN_WEBGL_URL}
    title={"hex tiles"}
    canFullscreen
  />
);
const BigfootMap = () => (
  <IFrame aspect={16 / 10} url={BIGFOOT_MAP_URL} title={"bigfoot sightings"} />
);

export type RouteDefinition = {
  path: string;
  Component: FC;
  children?: RouteDefinition[];
};

export const ROUTES: RouteDefinition[] = [
  { path: "", Component: Homepage },
  {
    path: "projects",
    Component: Projects,
    children: [
      {
        path: "figrid",
        Component: ProjectsFigrid,
      },
      {
        path: "pixijs",
        Component: Pixijs,
        children: [{ path: ":id", Component: Pixijs }],
      },
      { path: "babylonjs", Component: Babylonjs },
      { path: "breakout", Component: Breakout },
      { path: "platformer", Component: Platformer },
      { path: "hex_tiles", Component: HexTiles },
      { path: "webgl", Component: PlainWebGL },
      { path: "bigfoot", Component: BigfootMap },
      {
        path: "shaders",
        Component: Shaders,
        children: [{ path: ":category", Component: Shaders }],
      },
      {
        path: "sketches",
        Component: Sketches,
        children: [{ path: ":id", Component: Sketches }],
      },
    ],
  },
  { path: "photos", Component: Photos },
  // {
  //   path: "posts",
  //   Component: Posts,
  //   children: [{ path: ":id", Component: PostsContent }],
  // },
];

export const Routing = () => {
  const { pathname } = useLocation();

  useEffect(() => {
    const baseRoute = pathname.split("/")[1];
    document.title = baseRoute
      ? `${DOCUMENT_TITLE_ROOT} â€” ${baseRoute}`
      : DOCUMENT_TITLE_ROOT;
  }, [pathname]);

  const renderRoutes = (routes: RouteDefinition[], prefix = "") =>
    routes.reduce((renderedRoutes, { path, Component, children }) => {
      const fullPath = `${prefix}/${path}`;

      if (children) {
        renderedRoutes.push(...renderRoutes(children, fullPath));
      }

      return renderedRoutes.concat(
        <Route path={fullPath} element={<Component />} />
      );
    }, [] as ReactNode[]);

  return (
    <Suspense fallback={<Loader />}>
      <Routes>{renderRoutes(ROUTES)}</Routes>
    </Suspense>
  );
};
