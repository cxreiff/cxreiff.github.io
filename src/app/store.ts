import { createStore, applyMiddleware, compose } from 'redux'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { connectRouter, routerMiddleware } from 'connected-react-router'

import history from './history'
import homepage from '~/src/features/homepage/homepage.slice'

const router = connectRouter(history)

const middleware = [
    ...getDefaultMiddleware(),
    routerMiddleware(history),
];
 
if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger');
  middleware.push(logger);
}

const store = configureStore({
    reducer: {
        router,
        homepage,
    },
    middleware,
})

export type AppDispatch = typeof store.dispatch
export const useAppDispatch = () => useDispatch<AppDispatch>()

export type RootState = ReturnType<typeof store.getState>
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

export default store

