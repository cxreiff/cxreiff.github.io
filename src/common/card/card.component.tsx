import { FC } from 'react'
import cn from 'classnames'

import * as styles from './card.module.scss'

type CardProps = {
    onClick?: () => void,
    primary: string,
    secondary?: string,
    tertiary?: string,
}

export const Card: FC<CardProps> = ({ onClick, primary, secondary, tertiary }) => (
    <article
        className={cn(styles.card, { [styles.card_clickable]: onClick })}
        onClick={onClick}
    >
        <header>
            <h4>
                {primary}
            </h4>
        </header>
        {secondary && <footer>
            {secondary}
        </footer>}
        {tertiary && <p>
            {tertiary}
        </p>}
    </article>
)
