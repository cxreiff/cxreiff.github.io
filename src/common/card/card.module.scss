@use '../../shared.scss' as *;

.card {
    @include wide_card($accent-color-lighter, $accent-color-darker);
    display: flex;
    flex-wrap: wrap;

    img {
        width: 100%;
        margin: 0 0 1rem 0;
    }

    header {
        flex: 1 0;
        font-size: 1.2rem;
        
        h4 {
            margin: 0.5rem 0;
        }
    }

    footer {
        flex: 0 0 auto;
        font-size: 0.9rem;
        padding: 0.5rem;
    }

    p {
        flex: 0 0 100%;
        font-size: 0.9rem;
        border-top: 2px solid #BDBDBD;
        margin-top: 0;
        padding-top: 1rem;
    }
}

.card_clickable {
    @include interactive(
        0.4rem,
        $accent-color-darker,
        $accent-color-darkest,
        $accent-color-darker,
        $accent-color-darkest
    );
}
