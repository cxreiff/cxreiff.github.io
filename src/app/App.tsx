import React from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'

import 'normalize.css'

import store from './store'
import history from './history'

import Layout from '~/src/features/layout/layout.component'

const App: React.FC = () => (
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <Layout/>
        </ConnectedRouter>
    </Provider>
)

export default App
