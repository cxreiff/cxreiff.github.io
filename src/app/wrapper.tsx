import React, { ComponentType, ReactElement } from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'

import store from './store'
import history from './history'

const wrapper = (Component: ComponentType, props = {}): ReactElement => {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Component {...props} />
      </ConnectedRouter>
    </Provider>
  )
}

export default wrapper
