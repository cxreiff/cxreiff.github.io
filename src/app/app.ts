import { FC } from 'react'
import 'normalize.css'

import wrapper from './wrapper'
import Layout from '~/src/features/layout/layout.component'

const App: FC = () => wrapper(Layout)

export default App
