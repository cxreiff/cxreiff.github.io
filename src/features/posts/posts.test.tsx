import { mount, ReactWrapper } from 'enzyme'

import { Wrapper } from '~/src/app/wrapper'
import { Card } from '~/src/common/card/card.component'
import Posts from './posts.component'

const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}))

describe('posts', () => {

    let component: ReactWrapper<typeof Wrapper, void>

    beforeAll(() => {
        component = mount<typeof Wrapper, void>(
            <Wrapper>
                <Posts />
            </Wrapper>
        )
    })

    afterAll(() => {
        component.unmount()
    })

    afterEach(() => {
        jest.resetAllMocks()
    })

    it('should render', () => {
        expect(component.find(Posts)).toMatchSnapshot()
    })

    it('should navigate to post content', () => {
        component.find(Card).first().children().first().simulate('click')
        expect(mockNavigate).toHaveBeenCalled()
        expect(component.find(Posts)).toMatchSnapshot()
    })
})
