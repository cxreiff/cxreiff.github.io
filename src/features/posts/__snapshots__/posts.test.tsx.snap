// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts should navigate to post content 1`] = `
<Posts>
  <section>
    <PostsTile
      id="portfolio-site-design"
      key="portfolio-site-design"
      post={
        Object {
          "content": "
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam et ornare massa, vitae sagittis eros. Ut luctus risus ante, vel sollicitudin enim suscipit eget. Nam ac ipsum gravida, blandit dolor eget, eleifend ex. Aenean magna nibh, ultricies non erat ut, tempus congue nisi. Nunc felis elit, ornare a dictum in, elementum tincidunt justo. In aliquam vulputate odio, vel vestibulum enim fermentum et. Etiam eu aliquet leo.

\`\`\`tsx
const Component: FC = ({children}) => {
    return <div className={'test'}>
        {children}
    </div>
}
\`\`\`\`

Maecenas sagittis erat ut risus porttitor condimentum. Proin ut arcu nisl. Suspendisse maximus sodales luctus. Quisque accumsan pharetra purus id scelerisque. Nunc nunc nisi, suscipit quis fringilla in, hendrerit dignissim arcu. Sed turpis diam, malesuada ac metus ac, aliquam vulputate ante. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Maecenas sed ante fermentum, volutpat tellus eu, fringilla metus. Ut in mi ut leo aliquet dignissim in vitae mauris. Nam elementum lacus eu elit suscipit, eget blandit orci semper. Sed pretium dolor non varius placerat. Vestibulum non faucibus arcu, in consectetur ante. Nulla facilisi.

\`\`\`json
{
    'attribute1': ['a', 'b', 'c'],
    'attribute2': true,
    'attribute3': 12345
}
\`\`\`

Integer eget venenatis dui. Ut tellus orci, consequat vel aliquam vitae, condimentum in orci. Etiam volutpat bibendum lacinia. Aenean suscipit auctor magna ut pharetra. Duis nec lacus vel dui scelerisque finibus. Nam et fringilla ipsum, eget tempor dolor. Suspendisse et tristique lectus, sed blandit leo. Phasellus lacus ligula, tristique scelerisque augue id, finibus egestas enim. Aenean suscipit augue at ultricies venenatis. Vivamus imperdiet orci a dui rutrum hendrerit. Cras eget fringilla magna. In pellentesque erat nisl, non ullamcorper massa commodo sit amet. Etiam placerat id risus eget mollis. Curabitur commodo, urna eget fermentum aliquet, neque est blandit purus, eu gravida mi neque non augue.
",
          "data": Object {
            "date": 2021-05-31T00:00:00.000Z,
            "description": "Frameworks, structure, tradeoffs, and my development, testing, build, and deploy process.",
            "key": "portfolio-site-design",
            "title": "Portfolio site approach and design",
          },
          "excerpt": "",
          "isEmpty": false,
        }
      }
    >
      <Card
        onClick={[Function]}
        primary="Portfolio site approach and design"
        secondary="Mon May 31 2021"
        tertiary="Frameworks, structure, tradeoffs, and my development, testing, build, and deploy process."
      >
        <article
          className="card card_clickable"
          onClick={[Function]}
        >
          <header>
            <h4>
              Portfolio site approach and design
            </h4>
          </header>
          <footer>
            Mon May 31 2021
          </footer>
          <p>
            Frameworks, structure, tradeoffs, and my development, testing, build, and deploy process.
          </p>
        </article>
      </Card>
    </PostsTile>
    <PostsTile
      id="test-markdown"
      key="test-markdown"
      post={
        Object {
          "content": "
Second Test Markdown
===

Second test markdown.
",
          "data": Object {
            "date": 2022-01-11T00:00:00.000Z,
            "description": "Test markdown file.",
            "key": "test-markdown",
            "title": "Test Markdown File",
          },
          "excerpt": "",
          "isEmpty": false,
        }
      }
    >
      <Card
        onClick={[Function]}
        primary="Test Markdown File"
        secondary="Tue Jan 11 2022"
        tertiary="Test markdown file."
      >
        <article
          className="card card_clickable"
          onClick={[Function]}
        >
          <header>
            <h4>
              Test Markdown File
            </h4>
          </header>
          <footer>
            Tue Jan 11 2022
          </footer>
          <p>
            Test markdown file.
          </p>
        </article>
      </Card>
    </PostsTile>
  </section>
</Posts>
`;

exports[`posts should render 1`] = `
<Posts>
  <section>
    <PostsTile
      id="portfolio-site-design"
      key="portfolio-site-design"
      post={
        Object {
          "content": "
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam et ornare massa, vitae sagittis eros. Ut luctus risus ante, vel sollicitudin enim suscipit eget. Nam ac ipsum gravida, blandit dolor eget, eleifend ex. Aenean magna nibh, ultricies non erat ut, tempus congue nisi. Nunc felis elit, ornare a dictum in, elementum tincidunt justo. In aliquam vulputate odio, vel vestibulum enim fermentum et. Etiam eu aliquet leo.

\`\`\`tsx
const Component: FC = ({children}) => {
    return <div className={'test'}>
        {children}
    </div>
}
\`\`\`\`

Maecenas sagittis erat ut risus porttitor condimentum. Proin ut arcu nisl. Suspendisse maximus sodales luctus. Quisque accumsan pharetra purus id scelerisque. Nunc nunc nisi, suscipit quis fringilla in, hendrerit dignissim arcu. Sed turpis diam, malesuada ac metus ac, aliquam vulputate ante. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Maecenas sed ante fermentum, volutpat tellus eu, fringilla metus. Ut in mi ut leo aliquet dignissim in vitae mauris. Nam elementum lacus eu elit suscipit, eget blandit orci semper. Sed pretium dolor non varius placerat. Vestibulum non faucibus arcu, in consectetur ante. Nulla facilisi.

\`\`\`json
{
    'attribute1': ['a', 'b', 'c'],
    'attribute2': true,
    'attribute3': 12345
}
\`\`\`

Integer eget venenatis dui. Ut tellus orci, consequat vel aliquam vitae, condimentum in orci. Etiam volutpat bibendum lacinia. Aenean suscipit auctor magna ut pharetra. Duis nec lacus vel dui scelerisque finibus. Nam et fringilla ipsum, eget tempor dolor. Suspendisse et tristique lectus, sed blandit leo. Phasellus lacus ligula, tristique scelerisque augue id, finibus egestas enim. Aenean suscipit augue at ultricies venenatis. Vivamus imperdiet orci a dui rutrum hendrerit. Cras eget fringilla magna. In pellentesque erat nisl, non ullamcorper massa commodo sit amet. Etiam placerat id risus eget mollis. Curabitur commodo, urna eget fermentum aliquet, neque est blandit purus, eu gravida mi neque non augue.
",
          "data": Object {
            "date": 2021-05-31T00:00:00.000Z,
            "description": "Frameworks, structure, tradeoffs, and my development, testing, build, and deploy process.",
            "key": "portfolio-site-design",
            "title": "Portfolio site approach and design",
          },
          "excerpt": "",
          "isEmpty": false,
        }
      }
    >
      <Card
        onClick={[Function]}
        primary="Portfolio site approach and design"
        secondary="Mon May 31 2021"
        tertiary="Frameworks, structure, tradeoffs, and my development, testing, build, and deploy process."
      >
        <article
          className="card card_clickable"
          onClick={[Function]}
        >
          <header>
            <h4>
              Portfolio site approach and design
            </h4>
          </header>
          <footer>
            Mon May 31 2021
          </footer>
          <p>
            Frameworks, structure, tradeoffs, and my development, testing, build, and deploy process.
          </p>
        </article>
      </Card>
    </PostsTile>
    <PostsTile
      id="test-markdown"
      key="test-markdown"
      post={
        Object {
          "content": "
Second Test Markdown
===

Second test markdown.
",
          "data": Object {
            "date": 2022-01-11T00:00:00.000Z,
            "description": "Test markdown file.",
            "key": "test-markdown",
            "title": "Test Markdown File",
          },
          "excerpt": "",
          "isEmpty": false,
        }
      }
    >
      <Card
        onClick={[Function]}
        primary="Test Markdown File"
        secondary="Tue Jan 11 2022"
        tertiary="Test markdown file."
      >
        <article
          className="card card_clickable"
          onClick={[Function]}
        >
          <header>
            <h4>
              Test Markdown File
            </h4>
          </header>
          <footer>
            Tue Jan 11 2022
          </footer>
          <p>
            Test markdown file.
          </p>
        </article>
      </Card>
    </PostsTile>
  </section>
</Posts>
`;
